src/read/pe/export.rs

```diff
@@ -44,109 +47,255 @@ impl<'a> Debug for Export<'a> {
+    }
+
+    /// Returns the export address table entry at the given address index.
+    ///
+    /// This may be a local address, or the address of a forwarded export entry.
+    /// See [`Self::is_forward`] and [`Self::target_from_address`].
+    ///
+    /// `index` is a 0-based index into the export address table.
+    pub fn address_by_index(&self, index: u32) -> Result<u32> {
+        Ok(self
+            .addresses
+            .get(index as usize)
+            .read_error("Invalid PE export address index")?
+            .get(LE))
+    }
+
+    /// Returns the export address table entry at the given ordinal.
+    ///
+    /// This may be a local address, or the address of a forwarded export entry.
+    /// See [`Self::is_forward`] and [`Self::target_from_address`].
+    pub fn address_by_ordinal(&self, ordinal: u32) -> Result<u32> {
+        self.address_by_index(ordinal.wrapping_sub(self.ordinal_base()))
+    }
+
+    /// Returns the target of the export at the given address index.
+    ///
+    /// `index` is a 0-based index into the export address table.
+    pub fn target_by_index(&self, index: u32) -> Result<ExportTarget<'data>> {
+        self.target_from_address(self.address_by_index(index)?)
+    }
+
+    /// Returns the target of the export at the given ordinal.
+    pub fn target_by_ordinal(&self, ordinal: u32) -> Result<ExportTarget<'data>> {
+        self.target_from_address(self.address_by_ordinal(ordinal)?)
+    }
+
+    /// Convert an export address table entry into a target.
+    pub fn target_from_address(&self, address: u32) -> Result<ExportTarget<'data>> {
+        Ok(if let Some(forward) = self.forward_string(address)? {
+            let i = forward
+                .iter()
+                .position(|x| *x == b'.')
+                .read_error("Missing PE forwarded export separator")?;
+            let library = &forward[..i];
+            match &forward[i + 1..] {
+                [b'#', digits @ ..] => {
+                    let ordinal =
+                        parse_ordinal(digits).read_error("Invalid PE forwarded export ordinal")?;
+                    ExportTarget::ForwardByOrdinal(library, ordinal)
+                }
+                [] => {
+                    return Err(Error("Missing PE forwarded export name"));
+                }
+                name => ExportTarget::ForwardByName(library, name),
             }
+        } else {
+            ExportTarget::Address(address)
+        })
+    }
+
+    /// Return true if the export address table entry is a forward.
+    pub fn is_forward(&self, address: u32) -> bool {
+        let offset = address.wrapping_sub(self.virtual_address) as usize;
+        offset < self.data.len()
+    }
+
+    /// Return the forward string if the export address table entry is a forward.
+    pub fn forward_string(&self, address: u32) -> Result<Option<&'data [u8]>> {
+        let offset = address.wrapping_sub(self.virtual_address) as usize;
```

Comment:
why not reuse `is_forward` for these first two lines, that would de-duplicate them

Reply:
Because we need the offset value, which `is_forward` doesn't return. I've moved these lines into a new function.